lib/utils.js

```diff
@@ -63,25 +63,36 @@ function toFileURL(path) {
   return 'file://' + (isWin ? '/' : '') + path.replace(/\\/g, '/');
 }
 
-exports.getAlias = getAlias
+exports.getAlias = getAlias;
 function getAlias(loader, canonicalName) {
   var pluginIndex = loader.pluginFirst ? canonicalName.indexOf('!') : canonicalName.lastIndexOf('!');
+
   if (pluginIndex != -1)
     return getAlias(loader, canonicalName.substr(0, pluginIndex)) + '!' + getAlias(loader, canonicalName.substr(pluginIndex + 1));
 
   if (canonicalName.match(/\#[\:\{\?]/))
     throw new Error('Unable to alias the conditional dependency "' + canonicalName + '".');
 
+  var canonicalPath;
+
+  if (canonicalName.indexOf('/') >= 0)
+    canonicalPath = canonicalName.substr(canonicalName.indexOf('/'), canonicalName.length);
```

Comment:
This isn't quite right. map config can include `/` separators in the alias, so we need to do the path handling after the matching process.